lib/core/utils/preload-cssom.js

```diff
@@ -1,48 +1,113 @@
+/**
+ * Make an axios get request to fetch a given resource and resolve
+ * @method getExternalStylesheet
+ * @param {Object} arg an object with properties to configure the external XHR
+ * @property {Object} arg.resolve resolve callback on queue
+ * @property {Object} arg.reject reject callback on queue
+ * @property {String} arg.url string representing the url of the resource to load
+ * @property {Object} arg.rootNode document or shadowDOM root document for which to process CSSOM
+ * @property {Number} arg.timeout timeout to about network call
+ * @property {Function} arg.getStyleSheet a utility function to generate a style sheet for a given text content
+ * @property {String} arg.shadowId an id if undefined denotes that given root is a shadowRoot
+ * @property {Number} arg.priority css applied priority
+ * @private
+ */
+function getExternalStylesheet({
+	resolve,
+	reject,
+	url,
+	rootNode,
+	timeout,
+	getStyleSheet,
+	shadowId,
+	priority
+}) {
+	axe.imports
+		.axios({
+			method: 'get',
+			url,
+			timeout
+		})
+		.then(({ data }) => {
+			const sheet = getStyleSheet({
+				data,
+				isExternal: true,
+				shadowId,
+				root: rootNode,
+				priority
+			});
+			resolve(sheet);
+		})
+		.catch(reject);
+}
+
 /**
  * Returns a then(able) queue of CSSStyleSheet(s)
- * @param {Object} ownerDocument document object to be inspected for stylesheets
- * @param {number} timeout on network request for stylesheet that need to be externally fetched
- * @param {Function} convertTextToStylesheetFn a utility function to generate a style sheet from text
- * @return {Object} queue
+ * @method loadCssom
  * @private
+ * @param {Object} arg an object with projects essential to load CSSOM
+ * @property {Object} arg.rootNode document or shadowDOM root document for which to process CSSOM
+ * @property {Number} arg.rootIndex a number representing the index of the document or shadowDOM, used for priority
+ * @property {String} arg.shadowId an id if undefined denotes that given root is a shadowRoot
+ * @property {Number} timeout abort duration for network request
+ * @param {Function} getStyleSheet a utility function to generate a style sheet for a given text content
+ * @return {Object} queue
+
  */
-function loadCssom({ root, shadowId }, timeout, convertTextToStylesheetFn) {
+function loadCssom({ rootNode, rootIndex, shadowId, timeout, getStyleSheet }) {
```

Comment:
Can you please break this function up into more readable chunks? It's a 200 line function that does half a dozen different things.